### PR TITLE
Fix types, better exports and use tsup for build, remove parcel

### PR DESCRIPTION
**Pull request by sneridagh**
We are having problems in some setups in macos where `parcel` errors with a segmentation fault and we found also some weird errors related with it in `pnpm` lock, where the lock keeps updating because of it (apparently, their entries in the lock keep cycling infinitely). In addition, the resultant code has some weird names with a looong hash that end up in the dev tools. Then I also learned that `tsup` is more pure with their outputs and it was the recommended setup, see:

https://www.totaltypescript.com/how-to-create-an-npm-package

Totally worth it from top to bottom.

I also fixed the remaining types that forced us to `||true` in the build output. So overall this PR improves the current situation, in a non-breaking way.

### CODE REVIEW

File: packages/registry/package.json [hunk 1]
```diff
@@ -70,10 +63,9 @@
   },
   "scripts": {
     "watch": "parcel watch",
```
Comment (davisagli) [inline packages/registry/package.json:65]: There's still a parcel command here.

**Review by davisagli — COMMENTED** [review]
Could you explain more about the switch from parcel to tsup? What's the motivation and are there any tradeoffs?

**Comment by sneridagh** [comment]
@davisagli just added some more background in the description.

File: packages/registry/news/6461.bugfix [hunk 2]
```diff
@@ -0,0 +1 @@
+Fix types for addons TS, fix tsconfig for node side @sneridagh
```
Comment (stevepiercy) [inline packages/registry/news/6461.bugfix:1]:
```suggestion
Fix types for add-on's TypeScript. Fix `.tsconfig` for Node.js side. @sneridagh
```

File: packages/registry/news/6461.internal [hunk 3]
```diff
@@ -0,0 +1 @@
+Replace `parcel` with `tsup` for build @sneridagh
```
Comment (stevepiercy) [inline packages/registry/news/6461.internal:1]:
```suggestion
Replace `parcel` with `tsup` for build. @sneridagh
```

File: packages/scripts/news/6461.internal [hunk 4]
```diff
@@ -0,0 +1 @@
+Improve exports in @plone/registry, adapt the detection in i18n script @sneridagh
```
Comment (stevepiercy) [inline packages/scripts/news/6461.internal:1]:
```suggestion
Improve exports in `@plone/registry`. Adapt its detection in i18n script. @sneridagh
```

**Review by stevepiercy — CHANGES_REQUESTED** [review]
Was `parcel` something the developer directly called? If so, does its replacement merit more than a change log entry?

**Pull request merged**